app/src/main/java/com/davidtomas/taskyapp/core/domain/util/Result.kt

```diff
@@ -0,0 +1,31 @@
+package com.davidtomas.taskyapp.core.domain.util
+
+sealed class Result<T>(val data: T? = null, val error: Failure? = null) {
```

Comment:
Could it make sense to pass the failure type additionally here as a generic to get the type-safety?
```
sealed class Result<T, R : Failure>(val data: T? = null, val error: R? = null)
```
I think, then you should be able to create Results of specific failure types and your when-blocks will know about that when checking the error, e.g. 
```
val result = Result.Error<Unit, DataError>(DataError.Network.NO_INTERNET)
when(result.error) {
    // Here you know, result.error can only be of type DataResult 
}
```